pkg/appmgmt/general/general.go

```diff
@@ -123,20 +126,40 @@ func (os *Manager) getAppMetadata(pod *v1.Pod) (interfaces.ApplicationMetadata,
 	if err != nil {
 		log.Logger().Error("unable to get taskGroups by given pod", zap.Error(err))
 	}
+	ownerReferences := getOwnerReferences(pod)
+
 	placeholderTimeout, err := utils.GetPlaceholderTimeoutParam(pod)
 	if err != nil {
 		log.Logger().Warn("unable to get placeholder timeout by given pod.", zap.Error(err))
 	}
 	return interfaces.ApplicationMetadata{
-		ApplicationID: appId,
-		QueueName:     utils.GetQueueNameFromPod(pod),
-		User:          user,
-		Tags:          tags,
-		TaskGroups:    taskGroups,
+		ApplicationID:           appId,
+		QueueName:               utils.GetQueueNameFromPod(pod),
+		User:                    user,
+		Tags:                    tags,
+		TaskGroups:              taskGroups,
 		PlaceholderTimeoutInSec: placeholderTimeout,
+		OwnerReferences:         ownerReferences,
 	}, true
 }
 
+func getOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
```

Comment:
pls add comments to this function, to explain how we decide what owner-reference will be set for the placeholders.